pkg/manifests/external_dns.go

```diff
@@ -4,51 +4,68 @@ import (
 	"crypto/sha256"
 	"encoding/hex"
 	"encoding/json"
-	"path"
-
+	"fmt"
 	"github.com/Azure/aks-app-routing-operator/pkg/config"
 	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/go-autorest/autorest/to"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"path"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	provider        = "azure"
-	privateProvider = "azure-private-dns"
+	Provider        = "azure"
```

Comment:
if you are going to use these like an enum they should be a separate type.


```Go
type Provider int

const (
    Public Provider = iota
    Private
)

func (p Provider) String() string {
   switch p {
   case Public:
      return "azure"
   case Private:
      return "azure-private-dns"
   default:
       return ""
}
```

Reply:
This is how you should represent something like this in Go.

Reply:
done!